inbm/dispatcher-agent/dispatcher/sota/downloader.py

```diff
@@ -165,3 +168,66 @@ def download(self,
 
     def check_release_date(self, release_date: Optional[str]) -> bool:
         return self.is_valid_release_date(release_date)
+
+
+class TiberOSDownloader(Downloader):
+    """TiberOSDownloader class, child of Downloader
+
+       @param signature: signature used to preform signature check on downloaded image.
+    """
+
+    def __init__(self, signature: Optional[str] = None) -> None:
+        super().__init__()
+        self._signature = signature
+
+    def download(self,
+                 dispatcher_broker: DispatcherBroker,
+                 uri: Optional[CanonicalUri],
+                 repo: IRepo,
+                 username: Optional[str],
+                 password: Optional[str],
+                 release_date: Optional[str]) -> None:
+        """Downloads files and places image in local cache
+
+        @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+        @param uri: URI of the source location
+        @param repo: repository for holding the download
+        @param username: username to use for download
+        @param password: password to use for download
+        @param release_date: manifest release date
+        @raises SotaError: release date is not valid
+        """
+
+        if uri is None:
+            raise SotaError("URI is None while performing TiberOS download")
+
+        if password is None:
+            raise SotaError("JWT Token is None while performing TiberOS download")
```

Comment:
if uri and password are not optional, mark them as `str` instead of `Optional[str]`

also, it would be more consistent to log `password is None while performing TiberOS download` (rather than `JWT Token`)

Reply:
I think INBM can directly read the token from /etc/intel_edge_node/tokens/release-service/rs_access_token instead of requesting PUA to pass the token.